File: backend/src/main/java/tom/projects/monkeylog/model/exercise/ExerciseCategory.java
package tom.projects.monkeylog.model.exercise;

import java.util.List;

public enum ExerciseCategory {
    REPS("Reps", List.of(ExerciseType.REPS)),
    WEIGHTED("Weighted", List.of(ExerciseType.REPS, ExerciseType.WEIGHT)),
    DURATION("Duration", List.of(ExerciseType.TIME)),
    DISTANCE("Distance", List.of(ExerciseType.DISTANCE, ExerciseType.TIME));

    private final String name;
    private final List<ExerciseType> types;

    ExerciseCategory(String name, List<ExerciseType> types) {
        this.name = name;
        this.types = types;
    }

    public String getName() {
        return name;
    }

    public List<ExerciseType> getTypes() {
        return types;
    }
}

File: backend/src/main/kotlin/tom/com/workout/logbook/model/exercise/ExerciseCategory.kt
package tom.com.workout.logbook.model.exercise

import tom.com.workout.logbook.dto.statistics.StatisticsType

enum class ExerciseCategory(val statistics: List<StatisticsType>, val validFields: ValidFields) {
    REPS(listOf(StatisticsType.REPS), ValidFields(reps = true)),
    WEIGHTED(
        listOf(
            StatisticsType.WEIGHT,
            StatisticsType.REPS,
            StatisticsType.ONE_RM,
            StatisticsType.VOLUME
        ),
        ValidFields(reps = true, weight = true)
    ),
    DURATION(listOf(StatisticsType.TIME), ValidFields(time = true)),
    DISTANCE(
        listOf(StatisticsType.DISTANCE, StatisticsType.PACE, StatisticsType.TIME),
        ValidFields(distance = true, time = true)
    );

    data class ValidFields(
        val reps: Boolean = false,
        val weight: Boolean = false,
        val time: Boolean = false,
        val distance: Boolean = false,
    )
}
